api/src/accounts.rs

```diff
@@ -143,7 +143,7 @@ impl Account {
     }
 
     pub fn account(self) -> Result<impl Reply, Error> {
-        let account_state = self.account_state()?;
+        let account_state = self.account_state_impl()?;
         let account: AccountData = account_state
             .get_account_resource()?
```

Comment:
should directly get the `Account` resource, instead of all resources + modules under the account

Reply:
Good point - there are few more APIs that need to be changed to reflect fine-grained storage changes. How about we address this in a follow-up PR instead of increasing the scope of this PR?